### DISCUSSION TITLE
Test roles and permissions

### DISCUSSION THREAD

**Discussion by mohsenmojadam2019** [spatie/laravel-permission#1874] [Q&A]
hi

How can I do role and permissions  testing?

**Comment by erikn69** [comment]
```batch
composer install
./vendor/bin/phpunit
```

**Reply by mohsenmojadam2019** [reply]
True, but the package test folder is not created during installation and we can not access the tests written by the manufacturer

**Reply by mohsenmojadam2019** [reply]
https://github.com/spatie/laravel-permission/tree/main/tests

**Reply by PaolaRuby** [reply]
You have to clone this package, not run test from your app, if you want to test your app, you have to write your own testing

**Reply by erikn69** [reply]
#1894
